src/helpers/coordinates.ts

```diff
@@ -56,16 +63,52 @@ export function lettersToNumber(letters: string): number {
  */
 export function toCartesian(xc: string): Position {
   xc = xc.toUpperCase().trim();
-  const match = xc.match(cellReference);
-  if (match !== null) {
-    const [m, letters, numbers] = match;
-    if (m === xc) {
-      const col = lettersToNumber(letters);
-      const row = parseInt(numbers, 10) - 1;
-      return { col, row };
+  let numberPartStart: number | undefined = undefined;
+
+  // Note: looping by hand is uglier but ~2x faster than using a regex to match number/letter parts
+  for (let i = 0; i < xc.length; i++) {
+    const char = xc[i];
+
+    // Letter part
```

Comment:
I must say I'm confused by this comment. It says letter part, but mostly do things concerning numbers.
I think the comment would deserve to be more verbose to be more understandable by someone who didn't write this code :stuck_out_tongue: